### PR TITLE
update envoyproxy ratelimit version

### PR DESCRIPTION
**Pull request by johnzheng1975**
update envoyproxy ratelimit version, and disable DISABLE_STATS since it bring huge log

**Please provide a description of this PR:**

### CODE REVIEW

**Comment by johnzheng1975** [comment]
The version had been fully test on my prod env, with huge load.

**Comment by kfaseela** [comment]
cc @zirain

File: samples/ratelimit/rate-limit-service.yaml [hunk 1]
```diff
@@ -113,13 +113,15 @@ spec:
...
       containers:
-      - image: envoyproxy/ratelimit:9d8d70a8 # 2022/08/16
+      - image: envoyproxy/ratelimit:30a4ce1a # 2024/08/01
         imagePullPolicy: IfNotPresent
         name: ratelimit
         command: ["/bin/ratelimit"]
         env:
         - name: LOG_LEVEL
           value: debug
+        - name: DISABLE_STATS
```
Comment (zirain) [inline samples/ratelimit/rate-limit-service.yaml:123]: why disable stats？

**Comment by dhawton** [comment]
For samples, it is best to leave this on and allow people to disable it. This is meant for testing/demos/beginners, not production use.

**Comment by craigbox** [comment]
This should probably be referenced on https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/.
That page does say

> The statistics mentioned on the [Envoy rating limiting page](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/local_rate_limit_filter#statistics) are disabled by default. You can enable them with the following annotations during deployment:

Is this the same thing?

**Comment by zirain** [comment]
> This should probably be referenced on https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/. That page does say
> 
> > The statistics mentioned on the [Envoy rating limiting page](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/local_rate_limit_filter#statistics) are disabled by default. You can enable them with the following annotations during deployment:
> 
> Is this the same thing?

that doc said stats about envoy ratelimit filter, here we're talking envoy/ratelimit( a ratelimit server).

File: samples/ratelimit/rate-limit-service.yaml (hunk 1 above)
Comment (johnzheng1975) [reply samples/ratelimit/rate-limit-service.yaml:123]: Sorry not response in time.
This bring bring millions logs. Most custom do not why do not know how to disable stats.
Somehow these logs is useless, compare with debug log_level.

Comment (johnzheng1975) [reply samples/ratelimit/rate-limit-service.yaml:123]: https://github.com/envoyproxy/ratelimit/issues/764
FYI

**Comment by johnzheng1975** [comment]
I did not notice this, until it bring millions of logs each day on my prod env. 
This bring cost.

I guess most of user will not notice this like me, until loss money. ~_~
So I suggest to disable stats. After all, this is not so useful like debug log.

**Comment by johnzheng1975** [comment]
ok, let me revert change back as below:
```
         - name: DISABLE_STATS
-          value: "true"
+          value: "false"
```
You can change later if you think so.

**Review by zirain — APPROVED** [review]

**Comment by dhawton** [comment]
> I did not notice this parameter, until it bring millions of logs each day on my prod env. This bring cost.
> 
> I guess most of user will not notice this like me, until loss money. ^_^ So I suggest to disable stats by default. After all, this is not so useful like debug log.

Samples are not meant for prod environments.

**Pull request merged**